### ISSUE TITLE
setDefaultsOnInsert type broken in 8.1.0

### ISSUE THREAD

**Issue by richardsimko** [Automattic/mongoose#14282]
### Prerequisites

- [X] I have written a descriptive issue title
- [X] I have searched existing issues to ensure the bug has not already been reported

### Mongoose version

8.1.0

### Node.js version

20.x

### MongoDB server version

6.x

### Typescript version (if applicable)

5.3.x

### Description

The following code

```ts
await FooModel.updateOne(
          { foo: bar },
          { $set: {baz: true} },
          { setDefaultsOnInsert: true },
        );
```

Results in this type error

`Object literal may only specify known properties, and 'setDefaultsOnInsert' does not exist in type 'UpdateOptions & Omit<MongooseQueryOptions<AwardFare>, "lean">'.`

From what I can understand `setDefaultsOnInsert` hasn't been removed in the updated mongo driver version, and if it has this version of Mongoose should had been released as 9.0 and not 8.1.0.

This worked in 8.0.4

### Steps to Reproduce

See description

**Comment by vkarpov15** [comment]
I'm sorry for the inconvenience, this was a mistake that we fixed in #14278. We will ship 8.1.1 with the fix later today.

**Comment by richardsimko** [comment]
Thanks for the fast response!
